internal/validate/kube/kube.go

```diff
@@ -50,41 +57,81 @@ func Quiet() Option {
 	}
 }
 
+func Eks() Option {
+	eksConfig, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Printf("error while loading config: %s", err)
+	}
+
+	ec2Config, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Printf("error while loading config: %s", err)
+	}
+
+	return func(config *Config) {
+		config.eks = true
+		config.eksClient = eks.NewFromConfig(eksConfig)
+		config.ec2Client = ec2.NewFromConfig(ec2Config)
+	}
+}
+
 type validation func(ctx context.Context, config *Config) ([]validate.Result, error)
 
+type validationGroup struct {
+	Validate   validation
+	WaitMsg    string
+	SuccessMsg string
+	ErrMsg     string
+}
+
 // Validate will call a series of validation functions in a table driven tests style.
 func Validate(ctx context.Context, clientSet *kubernetes.Clientset, restConfig *rest.Config, opts ...Option) error {
-	config := &Config{
+	cfg := &Config{
 		namespace:  "default",
 		output:     os.Stdout,
 		exitStatus: false,
 		clientSet:  clientSet,
 		restConfig: restConfig,
+		eks:        false,
 	}
 
 	for _, opt := range opts {
-		opt(config)
+		opt(cfg)
 	}
 
-	log.SetOutput(config.output)
+	log.SetOutput(cfg.output)
 
-	var validations = []struct {
-		Validate   validation
-		WaitMsg    string
-		SuccessMsg string
-		ErrMsg     string
-	}{
-		{Pods, "validating pods", "pods validated", "validating pods failed"},
-		{Services, "validating services", "services validated", "validating services failed"},
-		{PVCs, "validating pvcs", "pvcs validated", "validating pvcs failed"},
-		{Connections, "validating connections", "connections validated", "validating connections failed"},
+	var validations []validationGroup
+	validations = append(validations, validationGroup{Pods, "validating pods", "pods validated", "validating pods failed"})
+	validations = append(validations, validationGroup{Services, "validating services", "services validated", "validating services failed"})
+	validations = append(validations, validationGroup{PVCs, "validating pvcs", "pvcs validated", "validating pvcs failed"})
+	validations = append(validations, validationGroup{Connections, "validating connections", "connections validated", "validating connections failed"})
+
+	if cfg.eks == true {
+		if !CurrentContextSetToEKSCluster() {
+			return fmt.Errorf("🛑 set current context to EKS cluster to use --eks flag")
```

Comment:
Use [Sourcegraph errors lib ](https://docs.sourcegraph.com/dev/background-information/languages/go_errors)here instead of `fmt.Errorf`

```suggestion
return errors.Newf("%s no EKS cluster detected in current context", validate.FailureEmoji)
```